### PR TITLE
Update emitter library and add support for ParametrizedUriEmitter

### PR DESCRIPTION
**Pull request by leventov**
- Update `emitter` library from 0.4.5 to 0.6.0
 - Move emitters from `io.druid.server.initialization` to the dedicated package
 - Add support for `ParametrizedUriEmitter` (introduced in emitter 0.5.0)
 - Support hierarchical properties in `JsonConfigurator` (was needed for `ParametrizedUriEmitter` configs)
 - Log created `RequestLogger`s, for some debugging/operations visibility.

### CODE REVIEW

File: server/src/test/java/io/druid/server/emitter/EmitterModuleTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    props.setProperty("druid.emitter.parametrized.httpEmitting.flushCount", "2");
+    props.setProperty("druid.emitter.parametrized.httpEmitting.basicAuthentication", "a:b");
+    props.setProperty("druid.emitter.parametrized.httpEmitting.batchingStrategy", "NEWLINES");
+    props.setProperty("druid.emitter.parametrized.httpEmitting.maxBatchSize", "4");
+    props.setProperty("druid.emitter.parametrized.httpEmitting.maxBufferSize", "8");
+    props.setProperty("druid.emitter.parametrized.httpEmitting.flushTimeOut", "1000");
+
+    final Emitter emitter =
+        makeInjectorWithProperties(props).getInstance(Emitter.class);
+    System.out.println(emitter);
```
[the remark this answers is not in the document]
Comment (leventov) [reply server/src/test/java/io/druid/server/emitter/EmitterModuleTest.java]: Thanks, fixed

File: api/src/main/java/io/druid/guice/JsonConfigurator.java [hunk 2]
```diff
@@ -165,6 +166,43 @@ public Message apply(String input)
...
+    }
+    if (dotIndex == 0) {
+      throw new ProvisionException(StringUtils.format("Double dot in property: %s", originalProperty));
+    }
+    if (dotIndex == property.length() - 1) {
+      throw new ProvisionException(StringUtils.format("Dot at the end of property: %s", originalProperty));
+    }
+    String nestedKey = property.substring(0, dotIndex);
+    Object nested = targetMap.computeIfAbsent(nestedKey, k -> new HashMap<String, Object>());
+    if (!(nested instanceof Map)) {
```
[the remark this answers is not in the document]
Comment (leventov) [reply api/src/main/java/io/druid/guice/JsonConfigurator.java]: The value may be a `Map` or a `String`, `!(nested instanceof Map)` checks that the nested map couldn't be populated

**Comment by leventov** [comment]
@himanshug could you please merge?

File: docs/content/configuration/index.md [hunk 3]
```diff
@@ -216,10 +216,26 @@ The Druid servers [emit various metrics](../operations/metrics.html) and alerts
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`druid.emitter.http.timeOut`|The timeout for data reads.|PT5M|
+|`druid.emitter.http.readTimeout`|The timeout for data reads.|PT5M|
 |`druid.emitter.http.flushMillis`|How often the internal message buffer is flushed (data is sent).|60000|
 |`druid.emitter.http.flushCount`|How many messages the internal message buffer can hold before flushing (sending).|500|
-|`druid.emitter.http.recipientBaseUrl`|The base URL to emit messages to. Druid will POST JSON to be consumed at the HTTP endpoint specified by this property.|none|
+|`druid.emitter.http.basicAuthentication`|Login and password for authentification in "login:password" form, e. g. `druid.emitter.http.basicAuthentication=admin:adminpassword`|not specified = no authentification|
```
Comment (drcrallen) [inline docs/content/configuration/index.md]: wish list: this to be a `PasswordProvider`

File: server/src/main/java/io/druid/server/log/LoggingRequestLoggerProvider.java [hunk 4]
```diff
@@ -41,6 +44,8 @@
   @Override
   public RequestLogger get()
   {
-    return new LoggingRequestLogger(mapper, setMDC, setContextMDC);
+    LoggingRequestLogger logger = new LoggingRequestLogger(mapper, setMDC, setContextMDC);
+    log.info(new Exception("Stack trace"), "Creating %s at", logger);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/log/LoggingRequestLoggerProvider.java]: should these be `debug` level?

**Comment by himanshug** [comment]
@leventov feel free to self merge when you're done.

**Pull request merged**